src/jump.jl

```diff
@@ -839,4 +841,164 @@ function MOI.set(::BilevelModel, ::MOI.UserCutCallback, func)
+
+function iterative_optimize!(solver, mode::ProductMode{T}, single_blm, model::BilevelModel, t0) where T
+    if isempty(mode.IterativeEpsilon)
+        return nothing
+    end
+
+    TerminationEpsilon = mode.epsilon
+
+    if MOI.get(solver, MOI.PrimalStatus()) in [FEASIBLE_POINT, NEARLY_FEASIBLE_POINT]
+        for (attr,val) in mode.IterativeAttributes
+            MOI.set(solver,MOI.RawOptimizerAttribute(attr),val)
+        end
+        println("Starting iterative ProductMode(), printing log below: ")
+        _print_iter_log(;Iteration=0, Regularization=mode.epsilon, TerminationStatus=MOI.get(solver, MOI.TerminationStatus()), PrimalStatus=MOI.get(solver, MOI.PrimalStatus()), ObjectiveValue=MOI.get(solver, MOI.ObjectiveValue()), t=time()-t0, upperline=true, header=true)
+        
+        if MOI.supports_incremental_interface(solver)
+            comp_idxs_in_solver = get_solver_comp_idxs(model, mode)
+            TerminationEpsilon = _iterative_optimize_solver!(solver, mode, comp_idxs_in_solver, t0)
+        else
+            TerminationEpsilon = _iterative_optimize_copy!(single_blm, solver, mode, model, t0)
+        end
+    else
+        @warn "Unable to start iterative ProductMode(). You may want to retry with a larger initial regularizer. For more information see below:"
+    end
+
+    _print_iter_log(;Iteration="Terminated", Regularization=TerminationEpsilon, TerminationStatus=MOI.get(solver, MOI.TerminationStatus()), PrimalStatus=MOI.get(solver, MOI.PrimalStatus()), ObjectiveValue=MOI.get(solver, MOI.ObjectiveValue()), t=time()-t0, upperline=true, header=true, bottomline=true)
+
+    return nothing
+
+end
+
+function get_solver_comp_idxs(model::BilevelModel, mode::ProductMode{T}) where T
+    [model.sblm_to_solver[comp_idx_in_sblm] for comp_idx_in_sblm in mode.comp_idx_in_sblm ]
+end
+
+function _iterative_optimize_solver!(solver, mode::ProductMode{T}, comp_idxs_in_solver::Vector{MathOptInterface.ConstraintIndex{F, S}}, t0) where {T, F, S}
+    for (iter,eps) in enumerate(mode.IterativeEpsilon)
+
+        SetIterPrimalStarts!(solver)
+        SetIterDualStarts!(solver)
+        SetIterRegularizations!(solver, eps, comp_idxs_in_solver, S)
+
+        MOI.optimize!(solver)
+
+        _print_iter_log(;Iteration="s$(iter)", Regularization=eps, TerminationStatus=MOI.get(solver, MOI.TerminationStatus()), PrimalStatus=MOI.get(solver, MOI.PrimalStatus()), ObjectiveValue=MOI.get(solver, MOI.ObjectiveValue()), t=time()-t0)
+
+        TerminationEpsilon = eps
+
+        if MOI.get(solver, MOI.PrimalStatus()) ∉ [FEASIBLE_POINT, NEARLY_FEASIBLE_POINT]
+            return TerminationEpsilon
+        end
+    end
+    return mode.IterativeEpsilon[end]
+end
+
+function _iterative_optimize_copy!(single_blm, solver, mode::ProductMode{T}, model::BilevelModel, t0) where T
+    for (iter,eps) in enumerate(mode.IterativeEpsilon)
+        SetIterPrimalStarts!(single_blm, solver, model.sblm_to_solver)
```

Comment:
in many functions and variable names this PR does nof follow the JuMP style guide: https://jump.dev/JuMP.jl/stable/developers/style/

Reply:
I'm still fairly inexperienced, so please excuse the question. Is this mostly about naming functions/variables like SetIterPrimalStarts! or IterEps? 
Or do I need to consider anything else?